filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReceiver.java

```diff
@@ -96,9 +96,12 @@ void addPart(int partId, byte [] part) {
             try {
                 Files.write(inprogressFile.toPath(), part, StandardOpenOption.WRITE, StandardOpenOption.APPEND);
             } catch (IOException e) {
-                log.log(Level.SEVERE, "Failed writing to file (" + inprogressFile.toPath() + "): " + e.getMessage(), e);
-                inprogressFile.delete();
-                throw new RuntimeException("Failed writing to file (" + inprogressFile.toPath() + "): ", e);
+                String message = "Failed writing to file (" + inprogressFile.toPath() + "): ";
+                log.log(Level.SEVERE, message + e.getMessage(), e);
```

Comment:
Consider catching the particular subclass of IOException thrown when the file already exists (FileAlreadyExistsException?), and only try deletion etc then.

Reply:
If we fail to write to the file we need to cleanup (we append to the file, so if it fails we have no idea of what has been written). So the change is just logging if we were unable to delete the file, we always need to delete no matter what the IOException was.

Reply:
👍